### PR TITLE
use config file instead of console menu

### PR DESCRIPTION
**Pull request by Kahdeg-15520487**
use ini config file.
the ini config file will be generated on first launch in bin/.
add custom option for add X resources and speed hack's multiplier

### CODE REVIEW

**Comment by rogeriocamorim** [comment]
What is the advantage of that? The console looks more user-friendly.
Also with the console maybe in the future we can have enable/disable on the go.

**Comment by rogeriocamorim** [comment]
Gotcha was not updated on the recent facts.

**Comment by Kahdeg-15520487** [comment]
> but console is what make we got detected easly,
i removed all the console stuff

> is this has been tested and work properly?..
i have tested it locally, would need to test more in depth later.
the custom value stuff work too.

> im at bed rn gonna look into it tmrw.
and it way past my bedtime here too.
i just throw it out asap so that other can also chime in

**Comment by gir489returns** [comment]
This doesn't address the root detection in that Game Guard is looking for the word hack.

**Comment by Kahdeg-15520487** [comment]
if it is just string stuff then i just replace the wording

**Comment by rogeriocamorim** [comment]
I would rename all SpeedHack (even variables) to SpeedModifier just in case.

**Comment by zDestinate** [comment]
I have one without the console. It still got detected.

**Comment by rogeriocamorim** [comment]
I got this code and still detect it.

**Comment by Kahdeg-15520487** [comment]
> I got this code and still detect it.

just finished a run with this code compiled, not detected

**Comment by noobLue** [comment]
> I got this code and still detect it.

compile in release mode not debug

**Comment by rogeriocamorim** [comment]
> > I got this code and still detect it.
> 
> compile in release mode not debug

q: Why debug mode detects?

**Comment by DeathWrench** [comment]
Made a PR for this PR that fixes some stuff
https://github.com/Kahdeg-15520487/hd2/pulls

**Comment by noobLue** [comment]
> > > I got this code and still detect it.
> > 
> > 
> > compile in release mode not debug
> 
> q: Why debug mode detects?

Debug mode includes bunch of extra information in the dll that release mode doesn't.

**Comment by gir489returns** [comment]
I'm getting code 1013 now no matter what I do. Fuck these fucking gooks.

File: version/dllmain.cpp [hunk 1]
```diff
@@ -8,685 +8,506 @@
...
+	ini["features"]["show_all_map_icons"] = gData.ShowAllMapIcons ? "1" : "0";
+	ini["features"]["no_stas_turret_overheat"] = gData.NoStasTurretOverHeat ? "1" : "0";
+	ini["features"]["shield_no_cd"] = gData.ShieldNoCD ? "1" : "0";
+	ini["features"]["jetpack_no_cd"] = gData.JetpackNoCD ? "1" : "0";
+	ini["features"]["all_stratagems"] = gData.AllStratagems ? "1" : "0";
+	ini["features"]["all_equipment"] = gData.AllEquipment ? "1" : "0";
+	ini["features"]["all_armor"] = gData.AllArmor ? "1" : "0";
+	ini["features"]["speed_modify"] = gData.Speedhack ? "1" : "0";
+	ini["features"]["speed_multiplier"] = std::to_string(gData.SpeedMultiplier);
+	ini["features"]["ohk"] = gData.OHK ? "1" : "0";
```
Comment (gir489returns) [inline version/dllmain.cpp:62]: All of these strings can be potentially looked for by GameGuard. XORing them does nothing, as that only affects the at-rest binary. Once the game loads this into memory, these strings will be in plain text. We should hash these instead.

File: version/dllmain.cpp [hunk 2]
```diff
@@ -8,685 +8,506 @@
...
+		gData.ShowAllMapIcons = ini["features"]["show_all_map_icons"] == "1";
+		gData.NoStasTurretOverHeat = ini["features"]["no_stas_turret_overheat"] == "1";
+		gData.ShieldNoCD = ini["features"]["shield_no_cd"] == "1";
+		gData.JetpackNoCD = ini["features"]["jetpack_no_cd"] == "1";
+		gData.AllStratagems = ini["features"]["all_stratagems"] == "1";
+		gData.AllEquipment = ini["features"]["all_equipment"] == "1";
+		gData.AllArmor = ini["features"]["all_armor"] == "1";
+		gData.Speedhack = ini["features"]["speed_modify"] == "1";
+		gData.SpeedMultiplier = ini["features"].has("speed_multiplier") ? std::stof(ini["features"]["speed_multiplier"]) : 2;
+		gData.OHK = ini["features"]["ohk"] == "1";
```
Comment (gir489returns) [inline version/dllmain.cpp:101]: This is beyond retarded. Either the INI class should return datatypes that we can template, or the INI class returns an object that we just cast to gData. FOH with that fucking C-style bullshit.

File: version/dllmain.cpp [hunk 3]
```diff
@@ -8,685 +8,506 @@
...
-            
-
-
-        }
-    }
-    printf(_XOR_("[Exit] Unload\n"));
-    FreeConsole();
-    FreeLibraryAndExitThread(g_hModule, 0);
-    return 0;
+	// Initialize the proxy for the DLL
```
Comment (gir489returns) [inline version/dllmain.cpp:109]: Massive whitespace change for no reason.

File: version/dllmain.cpp [hunk 4]
```diff
@@ -8,685 +8,506 @@
...
-    }break;
-    case DLL_THREAD_ATTACH:
-    case DLL_THREAD_DETACH:
-    case DLL_PROCESS_DETACH:
-        FreeConsole();
-        FreeLibraryAndExitThread(hModule, 0);
-        break;
-    }
-    return TRUE;
+	switch (ul_reason_for_call)
```
Comment (gir489returns) [inline version/dllmain.cpp:494]: Massive whitespace change for no reason.

**Review by gir489returns — CHANGES_REQUESTED** [review]
Garbage. I suggest @emoisback do his own, but if he wants this, it needs improvement.

**Comment by emoisback** [comment]
ok thanks ill rewrite this to use

something like this

 bool boolValue = config.GetValue<bool>("HashInfHealthWithKey");

**Comment by UnholyScribe** [comment]
so are we getting a update?? lol

**Comment by KingofEverythingg** [comment]
Yall have a discord?

**Comment by Kahdeg-15520487** [comment]
tks for code review, i will restart work for this again.

**Pull request closed without merging**